### PR TITLE
HPCC-13731 WorkunitsService does not always go via workunit.hpp

### PR DESCRIPTION
**Pull request by richardkchapman**
I suspect it's going straight to the XML to try to common up between dali and
sasha workunits, but the code that pulled the wu XML from Sasha was broken so
that feature never worked anyway. In the interests of
portability/maintainability I propose to remove the (broken) sasha capability.
With a bit of luck once Cassandra is in widespread use for workunits we won't
need to archive them any more anyway.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith This is still work in progress but would be worth you reviewing before you go

File: plugins/workunitservices/workunitservices.cpp [hunk 1]
```diff
@@ -636,27 +591,31 @@ WORKUNITSERVICES_API void wsWorkunitTimeStamps( ICodeContext *ctx, size32_t & __
...
-            fixedAppend(mb, 20, item, "@time");             
-            varAppend(mb, 1024, item, NULL);                
-        }
+    SCMStringBuffer s;
+    Owned<IConstWUExceptionIterator> exceptions = &wu->getExceptions();
+    ForEach(*exceptions)
+    {
+        IConstWUException &e = exceptions->query();
+        tmpu = (unsigned) e.getSeverity();
+        mb.append(sizeof(tmpu),&tmpu);
```
Comment (jakesmith) [inline plugins/workunitservices/workunitservices.cpp]: I know not new, but good time to change.
This is equivalent to just : mb.append(tmpu);  or
mb.append((unsigned)e.getSeverity());

File: plugins/workunitservices/workunitservices.cpp [hunk 2]
```diff
@@ -636,27 +591,31 @@ WORKUNITSERVICES_API void wsWorkunitTimeStamps( ICodeContext *ctx, size32_t & __
...
+        tmpi = (int) e.getExceptionCode();
+        mb.append(sizeof(tmpi),&tmpi);
```
Comment (jakesmith) [inline plugins/workunitservices/workunitservices.cpp]: similar to above comment

File: plugins/workunitservices/workunitservices.cpp [hunk 3]
```diff
@@ -636,27 +591,31 @@ WORKUNITSERVICES_API void wsWorkunitTimeStamps( ICodeContext *ctx, size32_t & __
...
+        e.getExceptionFileName(s);
+        fixedAppend(mb, 32, s.str(), s.length());
+        tmpu = (unsigned) e.getExceptionLineNo();
+        mb.append(sizeof(tmpu),&tmpu);
```
Comment (jakesmith) [inline plugins/workunitservices/workunitservices.cpp]: similar to above comment

**Comment by jakesmith** [comment]
@richardkchapman - just minor comments.

**Comment by richardkchapman** [comment]
This is still work in progress - I know it won't compile (and I am deliberately leaving it that way to make sure I remember to finish it). I Will close for now though

**Pull request closed without merging**